packages/fury-adapter-oas3-parser/lib/parser/oas/parseContactObject.js

```diff
@@ -0,0 +1,65 @@
+const R = require('ramda');
+const {
+  createInvalidMemberWarning,
+} = require('../annotations');
+const {
+  hasKey, isExtension,
+} = require('../../predicates');
+const parseObject = require('../parseObject');
+const parseString = require('../parseString');
+const pipeParseResult = require('../../pipeParseResult');
+
+const name = 'Contact Object';
+
+const parseMember = context => R.cond([
+  [hasKey('name'), parseString(context, name, false)],
+  [hasKey('url'), parseString(context, name, false)],
+  [hasKey('email'), parseString(context, name, false)],
+  [isExtension, () => new context.namespace.elements.ParseResult()],
+  [R.T, createInvalidMemberWarning(context.namespace, name)],
+]);
+
+/**
+ * Parse the OpenAPI 'Contact Object' (`#/info/contact`)
+ * @see http://spec.openapis.org/oas/v3.0.2#contact-object
+ * @returns ParseResult<Link>
+ * @private
+ */
+const parseContactObject = context => pipeParseResult(
+  context.namespace,
+  parseObject(context, name, parseMember(context)),
+  (contactObject) => {
+    const contactName = contactObject.get('name');
+    const contactUrl = contactObject.get('url');
+    const contactEmail = contactObject.get('email');
+    const links = [];
+
+    if (contactUrl) {
+      const link = new context.namespace.elements.Link();
+      link.relation = 'contact';
+      link.href = contactUrl;
+
+      if (contactName) {
+        link.title = contactName;
```

Comment:
As we are re-using the same element in the parse result (contentName `StringElement`) we need to `clone()` it so it isn't the exact same instance.

When we put the same element multiple times in the result it can cause problems with the `.freeze()` function on `ParseResult` as it expects the parse result to only be filled with unique element instances. What the method does is add `.parent` properties to each element so you can traverse the tree upwards. In this case it can cause the parent to be the wrong element on line 55 as it will be this link element.

https://github.com/refractproject/minim/blob/f5b19ae31ce69cf2ec5040bee2c9d8f1ab6660c5/lib/primitives/Element.js#L50

Reply:
wow, ok fix on next commit